app/workers/plos_details.rb

```diff
@@ -20,15 +20,33 @@ def perform(plos_paper)
                  "api_key=#{ENV.fetch('PLOS_API_KEY')}"
 
     detail_resp = Net::HTTP.get_response(URI.parse(detail_url))
+    # did we get a proper response?
+    unless detail_resp.is_a? Net::HTTPSuccess
+      # don't bother parsing broken or empty output
+      sleep 6
```

Comment:
Yes. We would need to make the worker throw an exception when the response is about the request limit being reached. It would then retry after a certain time. Do we know what exactly the rate lmit is? 10 per minute? Then let's just make the delay 30 seconds.